modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -2995,6 +2995,12 @@ private RestoreLogicalState applyLogicalUpdates(
                         GridCacheContext cacheCtx = cctx.cacheContext(cacheId);
 
                         if (cacheCtx != null) {
+                            // Index was destroyed before restart, mark it as invalid, so that we don't start
+                            // hew destruction task. Existance of this record means that
+                            // DurableBackgroundTask that destroys indexes is already in metastorage and
```

Comment:
> is already in metastorage

By the way, how exactly are we guaranteeing it? Do we write a durable background task before the rename record?

Reply:
We guarantee it now